Content.Server/GameObjects/Components/Power/PowerCellSlotComponent.cs

```diff
@@ -0,0 +1,244 @@
+        }
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            _cellContainer = ContainerManagerComponent.Ensure<ContainerSlot>("cellslot_cell_container", Owner, out _);
+        }
+
+        void IExamine.Examine(FormattedMessage message, bool inDetailsRange)
+        {
+            string sizeLetter = SlotSize switch
+            {
+                PowerCellSize.Small => Loc.GetString("S"),
+                PowerCellSize.Medium => Loc.GetString("M"),
+                PowerCellSize.Large => Loc.GetString("L"),
+                _ => "???"
+            };
+            if (DescFormatString != null) message.AddMarkup(string.Format(DescFormatString, sizeLetter));
+        }
+
+        /// <summary>
+        /// Remove the cell from this component. If a user is specified, the cell will be put in their hands
+        /// or failing that, at their feet. If no user is specified the cell will be put at the location of
+        /// the parent of this component.
+        /// </summary>
+        /// <param name="user">(optional) the user to give the removed cell to.</param>
+        /// <param name="playSound">Should <see cref="CellRemoveSound"/> be played upon removal?</param>
+        /// <returns>The cell component of the entity that was removed, or null if removal failed.</returns>
+        public PowerCellComponent? EjectCell(IEntity? user, bool playSound = true)
+        {
+            var cell = Cell;
+            if (cell == null || !CanRemoveCell) return null;
+            if (!_cellContainer.Remove(cell.Owner)) return null;
+            //Dirty();
+            if (user != null)
+            {
+                if (!user.TryGetComponent(out HandsComponent? hands) || !hands.PutInHand(cell.Owner.GetComponent<ItemComponent>()))
+                {
+                    cell.Owner.Transform.Coordinates = user.Transform.Coordinates;
+                }
+            }
+            else
+            {
+                cell.Owner.Transform.Coordinates = Owner.Transform.Coordinates;
+            }
+
+            if (playSound && CellRemoveSound != null)
+            {
+                EntitySystem.Get<AudioSystem>().PlayFromEntity(CellRemoveSound, Owner);
+            }
+            Owner.SendMessage(this, new PowerCellChangedMessage(true));
+            return cell;
+        }
+
+        /// <summary>
+        /// Tries to insert the given cell into this component. The cell will be put into the container of this component.
+        /// </summary>
+        /// <param name="cell">The cell to insert.</param>
+        /// <param name="playSound">Should <see cref="CellInsertSound"/> be played upon insertion?</param>
+        /// <returns>True if insertion succeeded; false otherwise.</returns>
+        public bool InsertCell(IEntity cell, bool playSound = true)
+        {
+            if (Cell != null) return false;
+            if (!cell.TryGetComponent<PowerCellComponent>(out var cellComponent)) return false;
```

Comment:
I prefer TryGetComponent(out T component) over TryGetComponent<T>(out var component) because it's shorter but that's just me.